azure-ts-dynamicresource/cdnCustomDomain.ts

```diff
@@ -0,0 +1,231 @@
+    }
+
+    const credentials = await msRestAzure.loginWithServicePrincipalSecret(clientID, clientSecret, tenantID);
+    return new cdnManagement.CdnManagementClient(credentials, subscriptionID);
+  }
+
+  async check(olds: CustomDomainOptions, news: CustomDomainOptions): Promise<pulumi.dynamic.CheckResult> {
+    const cdnClient = await this.getCDNManagementClient();
+    let result: pulumi.dynamic.CheckResult;
+
+    // If none of the CDN properties changed, then there is nothing to be validated.
+    if (olds.profileName === news.profileName &&
+        olds.endpointName === news.endpointName &&
+        olds.customDomainHostName === news.customDomainHostName) {
+      return {
+        inputs: news
+      };
+    }
+
+    const validationOutput = await cdnClient.endpoints.validateCustomDomain(
+      news.resourceGroupName, news.profileName, news.endpointName, news.customDomainHostName);
+    if (!validationOutput.customDomainValidated) {
+      result = {
+        failures: [
+          {
+            property: news.customDomainName,
+            reason: validationOutput.reason || "domain_validation_failed"
+          }
+        ]
+      };
+    } else {
+      result = {
+        inputs: news
+      };
+    }
+
+    return result;
+  }
+  async diff(id: string, olds: CustomDomainOutputs, news: CustomDomainOptions): Promise<pulumi.dynamic.DiffResult> {
+    const oldInputs = olds.inputs;
+    const replaces: string[] = [];
+    let changes = false;
+    let deleteBeforeReplace = false;
+
+    if (oldInputs.customDomainHostName !== news.customDomainHostName || oldInputs.customDomainName !== news.customDomainName) {
+      console.warn("Changing the domain name properties will cause a downtime.")
+
+      deleteBeforeReplace = true;
+      if (oldInputs.customDomainHostName !== news.customDomainHostName) {
+        replaces.push("customDomainHostName");
+      }
+      if (oldInputs.customDomainName !== news.customDomainName) {
+        replaces.push("customDomainHostName");
+      }
+    }
+
+    if (oldInputs.endpointName !== news.endpointName) {
+      deleteBeforeReplace = true;
+      replaces.push("endpointName");
+    }
+
+    // HTTPS can be enabled/disabled in-place.
+    if (oldInputs.httpsEnabled !== news.httpsEnabled) {
+      changes = true;
+      replaces.push("httpsEnabled");
+    }
+
+    return {
+      deleteBeforeReplace: deleteBeforeReplace,
+      replaces: replaces,
+      changes: changes
+    };
+  }
+
+  async create(inputs: CustomDomainOptions): Promise<pulumi.dynamic.CreateResult> {
+    const cdnClient = await this.getCDNManagementClient();
+    const result =
+    await cdnClient.customDomains
+      .create(
+        inputs.resourceGroupName,
+        inputs.profileName,
+        inputs.endpointName,
+        inputs.customDomainName,
+        inputs.customDomainHostName);
+    console.log(`Custom domain provisioning state is ${ result.resourceState }.`);
```

Comment:
Do we want to keep `console.log` in the examle?

Reply:
I'd like to since the custom domain and HTTPS provisioning can, in some cases, not be immediately active.